src/System.Net.HttpListener/tests/HttpListenerResponseTests.Cookies.cs

```diff
@@ -166,10 +166,28 @@ public static IEnumerable<object[]> Cookies_TestData()
             Assert.Equal("name3=value3", response.Headers["Set-Cookie"]);
             Assert.Null(response.Headers["Set-Cookie2"]);
 
-            string clientResponse = GetClientResponse(146);
+            string clientResponse = GetClientResponse(expectedLength:146);
             Assert.Contains($"\r\nSet-Cookie: name3=value3\r\n", clientResponse);
             Assert.DoesNotContain("Set-Cookie2", clientResponse);
         }
+  
+        [Fact]
+        public async Task Cookies_AddMultipleInHeader_ClientReceivesExpectedHeaders()
+        {
+            HttpListenerResponse response = await GetResponse();
+            response.Headers.Add("Set-Cookie", "name1=value1");
+            response.Headers.Add("Set-Cookie", "name2=value2");
+            response.Headers.Add("Set-Cookie", "name3=value3");
+            response.Headers.Add("Set-Cookie", "name4=value4");
+
+            response.Close();
+
+            string clientResponse = GetClientResponse(expectedLength:224);
+            Assert.Contains($"\r\nSet-Cookie: name1=value1\r\n", clientResponse);
```

Comment:
nit: using '$' interpolation isn't required here as there are no variables to evaluate in these strings.

Reply:
Good catch. Fixed throughout the whole file.